Runtime/Tracking/Collision/CollisionIgnorer.cs

```diff
@@ -0,0 +1,220 @@
+        {
+            Targets.ElementAdded.RemoveListener(OnTargetAdded);
+            Targets.ElementRemoved.RemoveListener(OnTargetRemoved);
+        }
+
+        /// <summary>
+        /// Responds to a <see cref="GameObject"/> being added to <see cref="Sources"/> and ignores all collisions against <see cref="Targets"/>.
+        /// </summary>
+        /// <param name="source">The source to ignore collisions from.</param>
+        protected virtual void OnSourceAdded(GameObject source)
+        {
+            ToggleCollisions(source, Sources, Targets, true);
+        }
+
+        /// <summary>
+        /// Responds to a <see cref="GameObject"/> being removed from <see cref="Sources"/> and resumes all collisions against <see cref="Targets"/>.
+        /// </summary>
+        /// <param name="source">The source to restore collisions with.</param>
+        protected virtual void OnSourceRemoved(GameObject source)
+        {
+            ToggleCollisions(source, Sources, Targets, false);
+        }
+
+        /// <summary>
+        /// Responds to a <see cref="GameObject"/> being added to <see cref="Targets"/> and ignores all collisions against <see cref="Sources"/>.
+        /// </summary>
+        /// <param name="target">The target to ignore collisions on.</param>
+        protected virtual void OnTargetAdded(GameObject target)
+        {
+            ToggleCollisions(target, Targets, Sources, true);
+        }
+
+        /// <summary>
+        /// Responds to a <see cref="GameObject"/> being removed from <see cref="Targets"/> and resumes all collisions against <see cref="Sources"/>.
+        /// </summary>
+        /// <param name="target">The target to restore collisions on.</param>
+        protected virtual void OnTargetRemoved(GameObject target)
+        {
+            ToggleCollisions(target, Targets, Sources, false);
+        }
+
+        /// <summary>
+        /// Sets the collision state between <see cref="Sources"/> and <see cref="Targets"/>.
+        /// </summary>
+        /// <param name="state">Whether to ignore collisions or not.</param>
+        protected virtual void ToggleCollisions(bool state)
+        {
+            foreach (GameObject source in Sources.SubscribableElements)
+            {
+                ToggleCollisions(source, Sources, Targets, state);
+            }
+        }
+
+        /// <summary>
+        /// Sets the collision state between the source and targets.
+        /// </summary>
+        /// <param name="source">The source to set the collision state on.</param>
+        /// <param name="sources">A collection of sources to check if the given <see cref="source"/> belongs to.</param>
+        /// <param name="targets">A collection of targets to set the collision state on.</param>
+        /// <param name="state">Whether to ignore collisions or not.</param>
+        protected virtual void ToggleCollisions(GameObject source, GameObjectObservableList sources, GameObjectObservableList targets, bool state)
```

Comment:
`targetColliders` will be cleared and filled again each time you call it, sorry I didn't explain how that API works (and it's undocumented as far as I can see 🙄). To solve it let's do this:
```c#
protected virtual void ToggleCollisions(GameObject source, GameObjectObservableList sources, GameObjectObservableList targets, bool state)
{
    if (source == null || (!state && sources.Contains(source)))
    {
        return;
    }

    source.GetComponentsInChildren(sourceColliders);

    foreach (GameObject target in targets.SubscribableElements)
    {
        if (target == null)
        {
            continue;
        }

        target.GetComponentsInChildren(targetColliders);

        foreach (Collider sourceCollider in sourceColliders)
        {
            foreach (Collider targetCollider in targetColliders)
            {
                Physics.IgnoreCollision(sourceCollider, targetCollider, state);
            }
        }
    }
}
```